charts/datadog/templates/_daemonset-volumes-linux.yaml

```diff
@@ -78,7 +78,7 @@
   name: debugfs
 - name: sysprobe-socket-dir
   emptyDir: {}
-{{- if and (or .Values.datadog.systemProbe.enableTCPQueueLength .Values.datadog.systemProbe.enableOOMKill) .Values.datadog.systemProbe.enableDefaultKernelHeadersPaths }}
+{{- if and (eq (include "runtime-compilation-enabled" .) "true") .Values.datadog.systemProbe.enableDefaultKernelHeadersPaths }}
```

Comment:
`.Values.datadog.systemProbe.enableRuntimeCompiler` should have been included here in the first place - this is just a small bug fix - which admittedly I should have put in a separate PR, I had just included it here because in my original set of changes I was already changing this line 😬  If you think it makes sense to address separately, I can cut a new PR for that